src/main/java/com/mraof/minestuck/blockentity/TransportalizerBlockEntity.java

```diff
@@ -212,21 +210,36 @@ private static boolean isDimensionForbidden(Level level)
 				|| typeKey.isPresent() && forbiddenDimTypes.contains(String.valueOf(typeKey.get().location()));
 	}
 	
+	public boolean hasId()
+	{
+		return !id.isEmpty();
+	}
+	
 	public String getId()
 	{
 		return id;
 	}
 	
-	public void setId(String id)
+	public void setId(String id, Player player)
```

Comment:
For packet security, I recommend returning with no change if the id has already been set instead of replacing it. Or if the transportalizer is locked.